tap_klaviyo/utils.py

```diff
@@ -118,16 +122,34 @@ def get_all_pages(source, url, api_key):
         else:
             break
 
+def list_info_request(url, api_key, id):
+    r = authed_get('list_info', url.format(list_id=id), {'api_key': api_key})
+    return r.json()
+
+def list_members_request(url, api_key, id, marker):
+    r = authed_get('list_members', url.format(list_id=id), {'api_key': api_key,
+                                                                'marker': marker})
+    return r.json()
+
+
 def get_list_members(url, api_key, id):
     marker = None
+    record_count = 0
     while True:
-        r = authed_get('list_members', url.format(list_id=id), {'api_key': api_key,
-                                                                'marker': marker})
-        response = r.json()
+        response = list_members_request(url, api_key, id, marker)
         if 'detail' in response.keys() and 'throttled' in response.get('detail'):
-            time.sleep(40)
-            continue
+            logger.info("request was throttled, entering retry logic")
+            response = None
+            retryLimit = 5
+            retryCount = 0
+            while(response == None and retryCount < retryLimit):
+                retryCount+=1
```

Comment:
just to keep it consistent 
```suggestion
                retryCount += 1
```